go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -0,0 +1,224 @@
+	ap.SupportsString(commitMessageArg, "m", "msg", "Use the given {{.LessThan}}msg{{.GreaterThan}} as the commit message.")
+	ap.SupportsFlag(allowEmptyFlag, "", "Allow recording a commit that has the exact same data as its sole parent. This is usually a mistake, so it is disabled by default. This option bypasses that safety.")
+	ap.SupportsString(dateParam, "", "date", "Specify the date used in the commit. If not specified the current system time is used.")
+	ap.SupportsFlag(forceFlag, "f", "Ignores any foreign key warnings and proceeds with the commit.")
+	ap.SupportsString(authorParam, "", "author", "Specify an explicit author using the standard A U Thor <author@example.com> format.")
+	return ap
+}
+
+// we are more permissive than what is documented.
+var supportedLayouts = []string{
+	"2006/01/02",
+	"2006/01/02T15:04:05",
+	"2006/01/02T15:04:05Z07:00",
+
+	"2006.01.02",
+	"2006.01.02T15:04:05",
+	"2006.01.02T15:04:05Z07:00",
+
+	"2006-01-02",
+	"2006-01-02T15:04:05",
+	"2006-01-02T15:04:05Z07:00",
+}
+
+func parseDate(dateStr string) (time.Time, error) {
+	for _, layout := range supportedLayouts {
+		t, err := time.Parse(layout, dateStr)
+
+		if err == nil {
+			return t, nil
+		}
+	}
+
+	return time.Time{}, errors.New("error: '" + dateStr + "' is not in a supported format.")
+}
+
+func parseAuthor(authorStr string) (string, string, error) {
+	if len(authorStr) == 0 {
+		return "", "", errors.New("Option 'author' requires a value")
+	}
+
+	reg := regexp.MustCompile("(?m)([^)]+) \\<([^)]+)") // Regex matches Name <email
+	matches := reg.FindStringSubmatch(authorStr)        // This function places the original string at the beginning of matches
+
+	// If name and email are provided
+	if len(matches) != 3 {
+		return "", "", errors.New("Author not formatted correctly. Use 'Name <author@example.com>' format")
+	}
+
+	name := matches[1]
+	email := strings.ReplaceAll(matches[2], ">", "")
+
+	return name, email, nil
+}
+
+// Trims the double quotes for the param.
+func trimQuotes(param string) string {
+	if len(param) > 0 && param[0] == '"' {
+		param = param[1:]
+	}
+
+	if len(param) > 0 && param[len(param)-1] == '"' {
+		param = param[:len(param)-1]
+	}
+
+	return param
+}
+
+func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// Get the information for the sql context.
+	dbName := ctx.GetCurrentDatabase()
+	dSess := sqle.DSessFromSess(ctx.Session)
+
+	doltdb, ok := dSess.GetDoltDB(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load %s", dbName)
+	}
+
+	rsr, ok := dSess.GetDoltDBRepoStateReader(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateReader", dbName)
+	}
+
+	rsw, ok := dSess.GetDoltDBRepoStateWriter(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateWriter", dbName)
+	}
+
+	ap := createArgParser()
+
+	// Get the args for DOLT_COMMIT.
+	args := make([]string, 1)
+	for i := range d.children {
+		temp := d.children[i].String()
```

Comment:
You don't want .String() here, you want to call child.Eval(ctx, row). See other functions for details.

Reply:
After you eval the child, you'll want to convert the value to a String using the sql package, something like sql.Longtext.Convert(val)